java/libphonenumber/src/com/google/i18n/phonenumbers/PhoneNumberUtil.java

```diff
@@ -3369,6 +3377,16 @@ public MatchType isNumberMatch(PhoneNumber firstNumberIn, PhoneNumber secondNumb
     return MatchType.NO_MATCH;
   }
 
+  // Return true if extension matches, possibly ignoring control characters
+  private boolean extensionMatches(String firstExtension, String secondExtension) {
+      if (firstExtension.equals(secondExtension)) {
+          return true;
+      }
+      String firstSanitized = firstExtension.replaceAll("[,;]", "");
```

Comment:
Performance: Each time we compare 2 phone numbers which have extensions, we'd hit this code where uncompiled regular expressions are used.

I think we should go with:
String firstSanitized = firstExtension.replace(',', '').replace(';', '');